### ISSUE TITLE
Configurable callback URL per SAML client

### ISSUE THREAD

**Issue by andrew-boutin** [jkacer/pac4j-extensions#2]
Does this support defining callback URLs per SAML client? I didn't see anything when looking through the source.

**Comment by jkacer** [comment]
Hi!
No, it doesn't. The callback URL is always `...../login?client_name=NAME`, where NAME is configured in the configuration table.

**Comment by andrew-boutin** [comment]
Let me know if this is harder than what I'm thinking, but wouldn't it be fairly simple to add an additional database column for `callbackUrl` so indirect clients could define their own. And then in the `DatabaseLoadedSAML2Clients` class in the `internalInit` function add `setCallbackUrl(clientCallbackUrlFromDatabase);` before `updateCallbackUrlOfIndirectClient(client);`? This would make it optional for users to not worry about setting a callbackUrl for each client if they didn't want to. It would default back to the original logic if no callbackUrl was present in the database row. Might need to add a function `getCallbackUrlIfPresent`in the `SamlClientDao` class.

Let me know what you think.

**Comment by andrew-boutin** [comment]
I did some experimenting and was able to get the `DatabaseLoadedSAML2ClientConfiguration` class to successfully load up a `callbackUrl` from my `SamlClientDao` class, however I discovered that I would need to modify the `Saml2Client` class to use the `callbackUrl` from the `DatabaseLoadedSAML2ClientConfiguration` class which would mean modifying one of the `pac4j` libraries outside of extensions.

Luckily I was able to set the `callbackUrl` on the `Saml2Client` in my custom `AbstractAuthenticationProcessingFilter`after the clients are loaded from the config but before the redirect happens. I was able to do this since the variable piece in the callback URL is the client name so I already had that information available from the `Saml2Client`.
